pandas/core/generic.py

```diff
@@ -2230,6 +2230,15 @@ def to_excel(
     ):
         df = self if isinstance(self, ABCDataFrame) else self.to_frame()
 
+        from pandas.io.excel._xlsxwriter import _XlsxWriter
```

Comment:
this is not the appropriate place for this, it needs to be in _XlsxWriter

Reply:
Need to disable merge_cell based on 1) if it is using _XlsxWriter and 2) if it has set constant_memory True. In _XlsxWriter, you cannot control `merge_cell`. I will try to move it this piece of logic to pandas\io\formats\excel.py

Reply:
I think placing it in `excel.py` is a better place than `generic.py` for starters.  We can then evaluate feasibility of moving `_XlsxWriter` if necessary, though your points are well-taken.

Reply:
Now the testing if using `_XlsxWriter` and `constant_memory==True` is moved to `excel.py`. 
And by referencing the solution of an [issue](https://github.com/jmcnamara/XlsxWriter/issues/292) of `xlsxwriter`, there is a perk in the updated code that we can keep merged cells even if the two aforementioned conditions are present.